module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -109,18 +110,42 @@ public function createEntity(): ResourceEntityInterface
     }
 
     /**
-     * Get a set of records that do not have metadata stored in the resource
-     * table.
+     * Get a set of records that are missing metadata in the resource table. If maxAge is specified, this includes also
+     * records that need to be updated.
+     *
+     * @param ?int    $lastId ID of last checked record, or null to start from beginning
+     * @param int     $limit  Limit for results
+     * @param ?int    $minAge Minimum age (in days) for metadata before it needs to be updated
+     * @param ?string $source Record source filter
      *
      * @return ResourceEntityInterface[]
      */
-    public function findMissingMetadata(): array
+    public function findMetadataToUpdate(?int $lastId, int $limit, ?int $minAge = null, ?string $source = null): array
     {
-        $dql = 'SELECT r '
-            . 'FROM ' . ResourceEntityInterface::class . ' r '
-            . "WHERE r.title = '' OR r.author IS NULL OR r.year IS NULL";
+        $dql = 'SELECT r FROM ' . ResourceEntityInterface::class . ' r';
+        $params = [];
+        if (null !== $minAge) {
+            $date = new DateTime();
+            $date->modify("-$minAge days");
```

Comment:
Any reason we can't do it like this to save a line (and use a less mutable style)?

```suggestion
            $date = new DateTime("now - $minAge days");
```